### PR TITLE
3.4.0: add SEConnectHelper, SECustomTabHelper

### PR DESCRIPTION
**Pull request by ConstantinKV**

### CODE REVIEW

File: app/src/main/java/com/saltedge/sdk/sample/features/ConnectActivity.java [hunk 1]
```diff
@@ -121,7 +120,7 @@ protected void onStart() {
     public void onConnectSessionSuccessStage(String connectionId, String connectionSecret, String rawJsonData) {
         PreferenceRepository.putConnectionSecret(connectionSecret);
         UITools.showShortToast(this, R.string.connection_connected);
-        closeActivity(true);
+        closeActivity(true, null);
```
Comment (mnewlive) [inline app/src/main/java/com/saltedge/sdk/sample/features/ConnectActivity.java:123]: Use named arguments please.

File: app/src/main/java/com/saltedge/sdk/sample/features/ConnectActivity.java [hunk 2]
```diff
@@ -203,14 +202,13 @@ private void setupActionBar() {
...
     private void requestConnectUrl() {
-        if (isRefreshMode()) {
-            createRefreshSessionUrl();
-        } else if (isReconnectViewMode()) {
-            boolean overrideCredentials = this.getIntent()
-                    .getBooleanExtra(Constants.KEY_OVERRIDE_CREDENTIALS, false);
-            createReconnectSessionUrl(overrideCredentials);
-        } else {
-            createConnectSessionUrl();
+        if (isRefreshMode()) {//Refresh connection
```
Comment (mnewlive) [inline app/src/main/java/com/saltedge/sdk/sample/features/ConnectActivity.java:205]: Maybe smth like `isRefreshConnectionMode()`, `isReconnecConnectiontViewMode`?
I think we don't need any extra comments here.

File: saltedge-library/src/main/java/com/saltedge/sdk/utils/SECustomTabHelper.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    private static final String[] chromePackages = new String[] { STABLE_PACKAGE, BETA_PACKAGE, DEV_PACKAGE, LOCAL_PACKAGE };
+    private static final String[] nonChromePackages = new String[] { "com.android.browser", "com.mi.globalbrowser", "org.mozilla.firefox", "org.mozilla.fennec", "com.opera.browser", "com.opera.mini.native", "com.microsoft.emmx", "com.yandex.browser" };
+    private static final String[] browserPackages = ArrayUtils.concat(chromePackages, nonChromePackages);
+    private static final List<String> chromePackagesList = Arrays.asList(chromePackages);
+    private static final List<String> browserPackagesList = Arrays.asList(browserPackages);
+
+    public static String getNonBrowserActivity(Context context, String url) {
+        PackageManager pm = context.getPackageManager();
+        int queryFlag = 0;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
```
Comment (mnewlive) [inline saltedge-library/src/main/java/com/saltedge/sdk/utils/SECustomTabHelper.java:59]: `android.os` move in import

File: saltedge-library/src/main/java/com/saltedge/sdk/utils/SECustomTabHelper.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    private static final String LOCAL_PACKAGE = "com.google.android.apps.chrome";
+    private static final String[] chromePackages = new String[] { STABLE_PACKAGE, BETA_PACKAGE, DEV_PACKAGE, LOCAL_PACKAGE };
+    private static final String[] nonChromePackages = new String[] { "com.android.browser", "com.mi.globalbrowser", "org.mozilla.firefox", "org.mozilla.fennec", "com.opera.browser", "com.opera.mini.native", "com.microsoft.emmx", "com.yandex.browser" };
+    private static final String[] browserPackages = ArrayUtils.concat(chromePackages, nonChromePackages);
+    private static final List<String> chromePackagesList = Arrays.asList(chromePackages);
+    private static final List<String> browserPackagesList = Arrays.asList(browserPackages);
+
+    public static String getNonBrowserActivity(Context context, String url) {
+        PackageManager pm = context.getPackageManager();
+        int queryFlag = 0;
```
Comment (mnewlive) [inline saltedge-library/src/main/java/com/saltedge/sdk/utils/SECustomTabHelper.java:58]: strange variable

File: app/src/main/java/com/saltedge/sdk/sample/features/ConnectActivity.java (hunk 1 above)
Comment (ConstantinKV) [reply app/src/main/java/com/saltedge/sdk/sample/features/ConnectActivity.java:123]: In Java no such term

**Pull request merged**